packages/block-editor/src/components/block-edit/test/edit.js

```diff
@@ -84,6 +86,47 @@ describe( 'Edit', () => {
 		expect( editElement ).toHaveClass( 'my-class' );
 	} );
 
+	it( 'should combine the default class name with a variation one', () => {
+		const edit = ( { className } ) => (
+			<div data-testid="foo-bar" className={ className } />
+		);
+
+		registerBlockType( 'core/test-block', {
+			edit,
+			save: noop,
+			category: 'text',
+			title: 'block title',
+			attributes: {
+				fruit: {
+					type: 'string',
+					default: 'Apples',
+				},
+			},
+		} );
+
+		registerBlockVariation( 'core/test-block', {
+			name: 'variation',
+			title: 'block variation title',
+			attributes: {
+				fruit: {
+					type: 'string',
+					default: 'Bananas',
+				},
+			},
+			isActive: ( { fruit } ) => fruit === 'Bananas',
```

Comment:
Nitpick, I think it's possible to [define variations during block registration](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-variations/#creating-a-block-variation), so it should be possible to drop the extra `registerBlockVariation` call (and hopefully also the `unregister`?)

```suggestion
		variations: [
			{
				name: 'variation',
				title: 'block variation title',
				attributes: {
					fruit: {
						type: 'string',
						default: 'Bananas',
					},
				},
				isActive: ( { fruit } ) => fruit === 'Bananas',
			}
		],
```

Reply:
(Not too important. If you think the separate registration reads better, feel free to keep it.)